sources/rest_api/README.md

```diff
@@ -0,0 +1,222 @@
+                        "resource": "pokemon_list",
+                        "field": "name",
+                    },
+                },
+            },
+        },
+    ],
+}
+
+pokemon_source = rest_api_source(pokemon_config)
+```
+Here a short summary:
+- The `client` node contains the base URL of the endpoints that we want to collect.
+- The `resources` which correspond to the API endpoints.
+
+We have a couple of simple resources (`berry` and `location`). The API endpoint is also the name of the dlt resource, and the name of the destination table. They don't need additional configuration.
+
+The next resource leverages some additional configuration. The endpoint `pokemon/` returns a list of pokemons, but it can be used also as `pokemon/{id or name}` to return a single pokemon. In this case we want the list, so we decided to rename the resource to `pokemon_list`, while the endpoint stays `pokemon/`. We do not specify the name of the destination table, so it will match the resource name.
+
+And now the `pokemon` one. This is actually a child endpoint of the `pokemon_list`: for each pokemon we want to get further details. So we need to make this resource a bit more smart, the endpoint `path` needs to be explicit, and we have to specify how the value of `name` will be resolved from another resource; this is actually telling the generic source that `pokemon` needs to be queried for each pokemon in `pokemon_list`.
+
+## Anatomy of the config object
+
+> **_TIP:_**  Import `RESTAPIConfig` from the `rest_api` module to have convenient tips.
+
+The config object passed to the REST API Generic Source has three main elements:
+
+```python
+my_config: RESTAPIConfig = {
+    "client": {
+        ...
+    },
+    "resource_defaults": {
+        ...
+    },
+    "resources": {
+        ...
+    ,
+}
+```
+
+
+
+`client` contains the configuration to connect to the APIs endpoints (e.g. base URL, authentication method, default behaviour for the paginator, and more).
+
+`resource_defaults` contains the default values to configure the dlt resources returned by this source.
+
+`resources` object contains the configuration for each resource. 
+
+The configuration with smallers scope will overwrite the one with the wider one:
+
+    Resource Configuration > Resource Defaults Configuration > Client Configuration
+
+## Reference
+
+### `client`
+
+#### `auth` [optional]
+Use the auth property to pass a token or a `HTTPBasicAuth` object for more complex authentication methods. Here some practical examples:
+
+1. Simple token (read from the `.dlt/secrets.toml` file):
+```python
+my_api_config: RESTAPIConfig = {
+    "client": {
+        "base_url": "https://my_api.com/api/v1/",
+        "auth": {
+            "token": dlt.secrets["sources.my_api.access_token"],
+        },
+    },
+    ...
+}
+```
+
+2. 
+```python
+from requests.auth import HTTPBasicAuth
+
+basic_auth = HTTPBasicAuth(dlt.secrets["sources.my_api.access_token"], "")

```

Comment:
I believe the basic auth is user/password: https://requests.readthedocs.io/en/latest/user/authentication/#basic-authentication

Reply:
Yes, we have this code running because Greenhouse uses only the user and colon and put that base64-encoded into basic auth. no password.

Reply:
I have replaced the token/nothing, with api_key/api_secret